UnityProject/Assets/Scripts/Managers/NetworkManagement/SpawnHandler.cs

```diff
@@ -11,13 +11,29 @@ public static class SpawnHandler
 
     public static void SpawnPlayer(NetworkConnection conn, short playerControllerId, JobType jobType = JobType.NULL)
     {
-        NetworkServer.AddPlayerForConnection(conn, CreatePlayer(jobType), playerControllerId);
-    }
+		GameObject player = CreatePlayer(jobType);
+		NetworkServer.AddPlayerForConnection(conn, player, playerControllerId);
+
+		Dictionary<string, GameObject> connectedPlayers = PlayerList.Instance.connectedPlayers;
+
+		//Notify all clients that connected players list should be updated
+		GameObject[] players = new GameObject[connectedPlayers.Count];
+		connectedPlayers.Values.CopyTo(players, 0);
+		UpdateConnectedPlayersMessage.Send(players);
+	}
 
     public static void RespawnPlayer(NetworkConnection conn, short playerControllerId, JobType jobType)
```

Comment:
Would you mind also commenting existing code you change next time around?

Reply:
I do add comments where its hard to understand what happens. In this example, "AddPlayerForConnection" is a unity function. I commented the code I added :)